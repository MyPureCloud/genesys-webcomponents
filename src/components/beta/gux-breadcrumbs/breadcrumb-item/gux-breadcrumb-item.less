@import (reference) '../../../../style/color.less';

gux-breadcrumb-item {
  .gux-breadcrumb-generation {
    display: flex;
    flex-direction: row;
    flex-wrap: nowrap;
    justify-content: flex-start;
    align-content: flex-start;
    align-items: center;

    .gux-breadcrumb-separator {
      margin: 0 8px;
      color: @gux-ghost-type;
    }

    gux-icon.gux-breadcrumb-separator {
      margin: 0 4px;
      width: 16px;
      height: 16px;
    }
  }
}
