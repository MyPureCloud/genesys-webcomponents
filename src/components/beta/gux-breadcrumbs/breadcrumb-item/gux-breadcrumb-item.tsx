import { Component, Element, h, JSX, Prop } from '@stencil/core';

@Component({
  styleUrl: 'gux-breadcrumb-item.less',
  tag: 'gux-breadcrumb-item'
})
export class GuxBreadcrumbItem {
  @Element()
  private root: HTMLGuxBreadcrumbItemElement;

  @Prop()
  href: string;

  private isLastBreadcrumb(): boolean {
    const parentNode = this.root.parentNode;
    const children = parentNode.children;

    return children[children.length - 1] === this.root;
  }

  private getBreadcrumb(): JSX.Element {
    if (this.href && !this.isLastBreadcrumb()) {
      return (
        <a class="gux-breadcrumb-content" href={this.href}>
          <slot />
        </a>
      );
    }

    return (
      <span class="gux-breadcrumb-content">
        <slot />
      </span>
    );
  }

  private getSeparatorIcon(): JSX.Element {
    if (this.isLastBreadcrumb()) {
      return null;
    }

    const container = this.root.closest('gux-breadcrumbs-beta');
    const accent = container.accent;

    switch (accent) {
      case 'primary':
        return (
          <span class="gux-breadcrumb-separator" aria-hidden="true">
            /
          </span>
        );
      case 'secondary':
        return (
          <gux-icon
            class="gux-breadcrumb-separator"
            icon-name="ic-chevron-small-right"
            decorative
          ></gux-icon>
        );
      default:
        return (
          <span class="gux-breadcrumb-separator" aria-hidden="true">
            /
          </span>
        );
    }
  }

  render(): JSX.Element {
    return (
      <span class="gux-breadcrumb-generation">
        {this.getBreadcrumb()}
        {this.getSeparatorIcon()}
      </span>
    );
  }
}
