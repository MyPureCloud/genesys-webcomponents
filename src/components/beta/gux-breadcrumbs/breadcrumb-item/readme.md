# gux-dropdown-option



<!-- Auto Generated Below -->


## Properties

| Property | Attribute | Description | Type     | Default     |
| -------- | --------- | ----------- | -------- | ----------- |
| `href`   | `href`    |             | `string` | `undefined` |


## Dependencies

### Depends on

- [gux-icon](../../../stable/gux-icon)

### Graph
```mermaid
graph TD;
  gux-breadcrumb-item --> gux-icon
  style gux-breadcrumb-item fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
