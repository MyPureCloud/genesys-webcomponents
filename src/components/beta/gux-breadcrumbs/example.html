<h1>Breadcrumbs</h1>

<h2>Primary</h2>

<gux-breadcrumbs-beta accent="primary">
    <gux-breadcrumb-item href="#">Department</gux-breadcrumb-item>
    <gux-breadcrumb-item>Category</gux-breadcrumb-item>
    <gux-breadcrumb-item href="#">Parent</gux-breadcrumb-item>
    <gux-breadcrumb-item>Current</gux-breadcrumb-item>
</gux-breadcrumbs-beta>

<h2>Secondary</h2>

<gux-breadcrumbs-beta accent="secondary">
    <gux-breadcrumb-item href="#">Department</gux-breadcrumb-item>
    <gux-breadcrumb-item>Category</gux-breadcrumb-item>
    <gux-breadcrumb-item href="#">Parent</gux-breadcrumb-item>
    <gux-breadcrumb-item>Current</gux-breadcrumb-item>
</gux-breadcrumbs-beta>
