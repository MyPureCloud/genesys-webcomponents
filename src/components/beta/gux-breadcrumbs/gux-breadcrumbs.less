@import (reference) '../../../style/color.less';

gux-breadcrumbs-beta {
  nav.breadcrumbs {
    display: block;
    border: none;
  }
  .separator {
    padding: 0 7px;
  }
  a.breadcrumb {
    color: -webkit-link;
    padding: 0;
  }
}

// Light theme
.gux-breadcrumbs-light-theme {
  .separator {
    color: @gux-type;
  }
}
.gux-light-theme {
  gux-breadcrumbs-beta {
    .gux-breadcrumbs-light-theme();
  }
}
