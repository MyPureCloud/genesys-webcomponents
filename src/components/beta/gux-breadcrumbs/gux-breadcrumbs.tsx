import { Component, Element, h, JSX, Prop } from '@stencil/core';

import { buildI18nForComponent, GetI18nValue } from '../../../i18n';

import breadcrumbsResources from './i18n/en.json';

export type GuxBreadcrumbAccent = 'primary' | 'secondary';

@Component({
  styleUrl: 'gux-breadcrumbs.less',
  tag: 'gux-breadcrumbs-beta'
})
export class GuxBreadcrumbs {
  private i18n: GetI18nValue;

  @Element()
  private root: HTMLGuxBreadcrumbsBetaElement;

  @Prop()
  accent: GuxBreadcrumbAccent = 'primary';

  async componentWillRender(): Promise<void> {
    this.i18n = await buildI18nForComponent(this.root, breadcrumbsResources);
  }

  render(): JSX.Element {
    return (
      <nav
        aria-label={this.i18n('breadcrumbs')}
        class={{
          'gux-breadcrumbs-container': true,
          'gux-large-text': this.accent === 'primary'
        }}
      >
        <slot />
      </nav>
    );
  }
}
