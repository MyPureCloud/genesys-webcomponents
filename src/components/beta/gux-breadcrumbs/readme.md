# gux-breadcrumbs



<!-- Auto Generated Below -->


## Properties

| Property | Attribute | Description | Type                       | Default     |
| -------- | --------- | ----------- | -------------------------- | ----------- |
| `accent` | `accent`  |             | `"primary" \| "secondary"` | `'primary'` |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
