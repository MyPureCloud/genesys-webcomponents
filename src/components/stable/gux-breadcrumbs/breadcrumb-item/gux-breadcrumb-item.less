@import (reference) '../../../../style/color.less';
@import (reference) '../../../../style/typography.less';

gux-breadcrumb-item {
  .gux-breadcrumb-generation {
    display: flex;
    flex-direction: row;
    flex-wrap: nowrap;
    align-content: flex-start;
    align-items: center;
    justify-content: flex-start;

    &.gux-primary {
      .gux-large-text();
    }

    .gux-breadcrumb-content {
      color: @gux-black-50;

      &.gux-link {
        color: @gux-blue-60;
      }
    }
    .gux-breadcrumb-separator {
      margin: 0 8px;
      color: @gux-grey-10;
    }

    gux-icon.gux-breadcrumb-separator {
      width: 16px;
      height: 16px;
      margin: 0 4px;
    }
  }
}
