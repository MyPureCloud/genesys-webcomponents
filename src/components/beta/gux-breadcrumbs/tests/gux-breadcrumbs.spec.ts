import { newSpecPage } from '@stencil/core/testing';

import { GuxBreadcrumbs } from '../gux-breadcrumbs';
import { GuxBreadcrumbItem } from '../breadcrumb-item/gux-breadcrumb-item';

const components = [GuxBreadcrumbs, GuxBreadcrumbItem];
const language = 'en';

describe('gux-breadcrumbs-beta', () => {
  let component: GuxBreadcrumbs;

  beforeEach(async () => {
    const page = await newSpecPage({
      components,
      html: `<gux-breadcrumbs-beta></gux-breadcrumbs-beta>`,
      language
    });

    component = page.rootInstance;
  });

  it('should build', async () => {
    expect(component).toBeInstanceOf(GuxBreadcrumbs);
  });

  describe('#render', () => {
    [
      `<gux-breadcrumbs-beta></gux-breadcrumbs-beta>`,
      `
        <gux-breadcrumbs-beta>
          <gux-breadcrumb-item>Category</gux-breadcrumb-item>
          <gux-breadcrumb-item href="#">Parent</gux-breadcrumb-item>
          <gux-breadcrumb-item>Current</gux-breadcrumb-item>
        </gux-breadcrumbs-beta>
      `,
      `
        <gux-breadcrumbs-beta>
          <gux-breadcrumb-item>Category</gux-breadcrumb-item>
          <gux-breadcrumb-item href="#">Parent</gux-breadcrumb-item>
          <gux-breadcrumb-item href="#">Current</gux-breadcrumb-item>
        </gux-breadcrumbs-beta>
      `,
      `
        <gux-breadcrumbs-beta accent="primary">
          <gux-breadcrumb-item>Category</gux-breadcrumb-item>
          <gux-breadcrumb-item href="#">Parent</gux-breadcrumb-item>
          <gux-breadcrumb-item>Current</gux-breadcrumb-item>
        </gux-breadcrumbs-beta>
      `,
      `
        <gux-breadcrumbs-beta accent="secondary">
          <gux-breadcrumb-item>Category</gux-breadcrumb-item>
          <gux-breadcrumb-item href="#">Parent</gux-breadcrumb-item>
          <gux-breadcrumb-item>Current</gux-breadcrumb-item>
        </gux-breadcrumbs-beta>
      `,
      `
        <gux-breadcrumbs-beta accent="invalid">
          <gux-breadcrumb-item>Category</gux-breadcrumb-item>
          <gux-breadcrumb-item href="#">Parent</gux-breadcrumb-item>
          <gux-breadcrumb-item>Current</gux-breadcrumb-item>
        </gux-breadcrumbs-beta>
      `
    ].forEach((html, index) => {
      it(`should render component as expected (${index + 1})`, async () => {
        const page = await newSpecPage({ components, html, language });

        expect(page.root).toMatchSnapshot();
      });
    });
  });
});
