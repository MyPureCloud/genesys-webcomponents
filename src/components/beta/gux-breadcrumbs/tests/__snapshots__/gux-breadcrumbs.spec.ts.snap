// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-breadcrumbs-beta #render should render component as expected (1) 1`] = `
<gux-breadcrumbs-beta>
  <!---->
  <nav aria-label="Breadcrumbs" class="gux-breadcrumbs-container"></nav>
</gux-breadcrumbs-beta>
`;

exports[`gux-breadcrumbs-beta #render should render component as expected (2) 1`] = `
<gux-breadcrumbs-beta>
  <!---->
  <nav aria-label="Breadcrumbs" class="gux-breadcrumbs-container">
    <gux-breadcrumb-item>
      <!---->
      <span class="gux-breadcrumb-generation">
        <span class="gux-breadcrumb-content">
          Category
        </span>
        <span aria-hidden="true" class="gux-breadcrumb-separator">
          /
        </span>
      </span>
    </gux-breadcrumb-item>
    <gux-breadcrumb-item href="#">
      <!---->
      <span class="gux-breadcrumb-generation">
        <a class="gux-breadcrumb-content" href="#">
          Parent
        </a>
        <span aria-hidden="true" class="gux-breadcrumb-separator">
          /
        </span>
      </span>
    </gux-breadcrumb-item>
    <gux-breadcrumb-item>
      <!---->
      <span class="gux-breadcrumb-generation">
        <span class="gux-breadcrumb-content">
          Current
        </span>
      </span>
    </gux-breadcrumb-item>
  </nav>
</gux-breadcrumbs-beta>
`;

exports[`gux-breadcrumbs-beta #render should render component as expected (3) 1`] = `
<gux-breadcrumbs-beta>
  <!---->
  <nav aria-label="Breadcrumbs" class="gux-breadcrumbs-container">
    <gux-breadcrumb-item>
      <!---->
      <span class="gux-breadcrumb-generation">
        <span class="gux-breadcrumb-content">
          Category
        </span>
        <span aria-hidden="true" class="gux-breadcrumb-separator">
          /
        </span>
      </span>
    </gux-breadcrumb-item>
    <gux-breadcrumb-item href="#">
      <!---->
      <span class="gux-breadcrumb-generation">
        <a class="gux-breadcrumb-content" href="#">
          Parent
        </a>
        <span aria-hidden="true" class="gux-breadcrumb-separator">
          /
        </span>
      </span>
    </gux-breadcrumb-item>
    <gux-breadcrumb-item href="#">
      <!---->
      <span class="gux-breadcrumb-generation">
        <span class="gux-breadcrumb-content">
          Current
        </span>
      </span>
    </gux-breadcrumb-item>
  </nav>
</gux-breadcrumbs-beta>
`;

exports[`gux-breadcrumbs-beta #render should render component as expected (4) 1`] = `
<gux-breadcrumbs-beta accent="primary">
  <!---->
  <nav aria-label="Breadcrumbs" class="gux-breadcrumbs-container">
    <gux-breadcrumb-item>
      <!---->
      <span class="gux-breadcrumb-generation">
        <span class="gux-breadcrumb-content">
          Category
        </span>
        <span aria-hidden="true" class="gux-breadcrumb-separator">
          /
        </span>
      </span>
    </gux-breadcrumb-item>
    <gux-breadcrumb-item href="#">
      <!---->
      <span class="gux-breadcrumb-generation">
        <a class="gux-breadcrumb-content" href="#">
          Parent
        </a>
        <span aria-hidden="true" class="gux-breadcrumb-separator">
          /
        </span>
      </span>
    </gux-breadcrumb-item>
    <gux-breadcrumb-item>
      <!---->
      <span class="gux-breadcrumb-generation">
        <span class="gux-breadcrumb-content">
          Current
        </span>
      </span>
    </gux-breadcrumb-item>
  </nav>
</gux-breadcrumbs-beta>
`;

exports[`gux-breadcrumbs-beta #render should render component as expected (5) 1`] = `
<gux-breadcrumbs-beta accent="secondary">
  <!---->
  <nav aria-label="Breadcrumbs" class="gux-breadcrumbs-container">
    <gux-breadcrumb-item>
      <!---->
      <span class="gux-breadcrumb-generation">
        <span class="gux-breadcrumb-content">
          Category
        </span>
        <gux-icon class="gux-breadcrumb-separator" decorative="" icon-name="ic-chevron-small-right"></gux-icon>
      </span>
    </gux-breadcrumb-item>
    <gux-breadcrumb-item href="#">
      <!---->
      <span class="gux-breadcrumb-generation">
        <a class="gux-breadcrumb-content" href="#">
          Parent
        </a>
        <gux-icon class="gux-breadcrumb-separator" decorative="" icon-name="ic-chevron-small-right"></gux-icon>
      </span>
    </gux-breadcrumb-item>
    <gux-breadcrumb-item>
      <!---->
      <span class="gux-breadcrumb-generation">
        <span class="gux-breadcrumb-content">
          Current
        </span>
      </span>
    </gux-breadcrumb-item>
  </nav>
</gux-breadcrumbs-beta>
`;

exports[`gux-breadcrumbs-beta #render should render component as expected (6) 1`] = `
<gux-breadcrumbs-beta accent="invalid">
  <!---->
  <nav aria-label="Breadcrumbs" class="gux-breadcrumbs-container">
    <gux-breadcrumb-item>
      <!---->
      <span class="gux-breadcrumb-generation">
        <span class="gux-breadcrumb-content">
          Category
        </span>
        <span aria-hidden="true" class="gux-breadcrumb-separator">
          /
        </span>
      </span>
    </gux-breadcrumb-item>
    <gux-breadcrumb-item href="#">
      <!---->
      <span class="gux-breadcrumb-generation">
        <a class="gux-breadcrumb-content" href="#">
          Parent
        </a>
        <span aria-hidden="true" class="gux-breadcrumb-separator">
          /
        </span>
      </span>
    </gux-breadcrumb-item>
    <gux-breadcrumb-item>
      <!---->
      <span class="gux-breadcrumb-generation">
        <span class="gux-breadcrumb-content">
          Current
        </span>
      </span>
    </gux-breadcrumb-item>
  </nav>
</gux-breadcrumbs-beta>
`;
